src/pages/en/guides/deploy/github.md

```diff
@@ -76,5 +76,8 @@ You can deploy an Astro site to GitHub Pages by using [GitHub Actions](https://g
 3. Commit the new workflow file and push it to GitHub.
 4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.
 5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.
+6. (Optionally) Setup a custom domain by adding `./public/CNAME` file with `sub.mydomain.com` as content.  
+    `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment.  
+    Don't forget to [setup DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).
 
```

Comment:
``````suggestion
6. Optional: Set up a custom domain by adding the following `./public/CNAME` file to your project: 
```
sub.mydomain.com
````
`actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
``````
Hey @JulianCataldo thanks so much for stepping up for this! 🙌 

Looks great, I'm just thinking of maybe a way to separate a direct, explicit instruction from the under-the-hood explanation as clearly as possible. And, when a new file must be created, if we can figure out a way to do it sensibly and make it look nice, we do like to provide a code sample. 

Also, line breaks to start a new sentence without a blank line between them don't look the best in our lists on the Docs site, I find. So, we try to structure things where a blank line between paragraphs makes sense/looks good, or keep everything in one paragraph (again, if it doesn't look terrible). Welcome to Docs: not just getting the content right, but sometimes changing content to fit the visuals!

I can't see a preview of this while I'm just reviewing the file diff on GitHub's website, so this isn't a set in stone suggestion for change. Just wondering if some kind of tweaking like this might give us a cleaner look on the site. What do you think? Do you mind experimenting?

Reply:
You're welcome :)

Under-the-hood things might be totally superfluous.  
People just want to get the job done ;p
OK for a second version with an attempt at better spacings.

About "(Optionally)" I've copied this from another place in docs, for coherence sake. BTW, I originally came with "(Optional)" as a first guess.

Also, I think that:

> Blabla: some blaba: more blah.

(Two colons in the same sentence) is syntactically incorrect, in English / French…

![Screenshot 2022-07-25 at 4 00 42](https://user-images.githubusercontent.com/603498/180678788-4b698ef9-8a60-4bd0-a7c8-8b6265d5ee48.png)

Reply:
Yes, if adding the file should really be done earlier, then I think moving it up sounds good! 

How important/helpful would you rank this info:  "`actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment."  This is what I was wondering about re: maybe unnecessary info for what are otherwise pretty short steps? 

And yes, let's keep the optionally consistent then. Those integrations docs were all written separately, and the content is mirrored in from an external source on the astro repo, so I sometimes forget to consider their style, too. Great catch!

Reply:
Mhhh, I'm not quite accustomed to this "hot-edit" workflow, I can't find a way to (re)edit the current PR / concerned files 😅

Reply:
Ooh, one thing I do is just, on this PR on the GitHub.com itself, is there's a tab at the top for "Files Changed."  (Right now, we're in "Conversation.")

If you click to see the files changed, then as you hover over lines along the gutter, you should see a big PLUS sign visible. If you click and hold and drag down, you can select multiple lines to edit.  It will look like you're just making a comment at first, but if you press the icon that looks like a document with a plus sign in the middle, you'll actually be able to "suggest" edits to what's there. So, you don't get a nice preview, but you can literally change files that way and make "suggestions" and evne commit them that way.

Let me know if trying that doesn't work for you!